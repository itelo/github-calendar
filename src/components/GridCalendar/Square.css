.square {
  height: 8px;
  width: 8px;
  background: rgb(235, 237, 240);
}

.big-square {
  height: 10px;
  width: 10px;
  background: rgb(235, 237, 240);
}

.square-container {
  margin-top: 2px;
  margin-left: 2px;
}

.nothing {
  background: #ebedf0;
}

.little {
  background: #c6e48b;
}
.moderate {
  background: #7bc96f;
}
.alot {
  background: #239a3b;
}
.most {
  background: #196127;
}
