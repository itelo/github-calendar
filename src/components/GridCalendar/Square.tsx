import React from "react";
import { getColorClass } from "../../utils/squares";
import "./Square.css";

const Square = (props: { count: number }) => (
  <div className="square-container">
    <div
      data-testid="square-color"
      className={`square ${getColorClass(props.count)}`}
    />
  </div>
);

export default Square;
