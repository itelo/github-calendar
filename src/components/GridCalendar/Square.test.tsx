import React from "react";
import { render, fireEvent } from "react-testing-library";
import "react-testing-library/cleanup-after-each";
import Square from "./Square";
import "jest-dom/extend-expect";
import {
  CONTRIBUTIONS_MOST,
  CONTRIBUTIONS_ALOT,
  CONTRIBUTIONS_MODERATE,
  CONTRIBUTIONS_LITTLE
} from "../../utils/defines";

describe("<Square />", () => {
  test("test square passing CONTRIBUTIONS_MOST, should have class most", () => {
    const { getByTestId, container } = render(
      <Square count={CONTRIBUTIONS_MOST} />
    );
    const square = getByTestId("square-color");

    expect(square).toHaveClass("most");

    expect(container).toMatchSnapshot();
  });

  test("test square passing CONTRIBUTIONS_ALOT, should have class alot", () => {
    const { getByTestId, container } = render(
      <Square count={CONTRIBUTIONS_ALOT} />
    );
    const square = getByTestId("square-color");

    expect(square).toHaveClass("alot");

    expect(container).toMatchSnapshot();
  });

  test("test square passing CONTRIBUTIONS_MODERATE, should have class moderate", () => {
    const { getByTestId, container } = render(
      <Square count={CONTRIBUTIONS_MODERATE} />
    );
    const square = getByTestId("square-color");

    expect(square).toHaveClass("moderate");

    expect(container).toMatchSnapshot();
  });

  test("test square passing CONTRIBUTIONS_LITTLE, should have class little", () => {
    const { getByTestId, container } = render(
      <Square count={CONTRIBUTIONS_LITTLE} />
    );
    const square = getByTestId("square-color");

    expect(square).toHaveClass("little");

    expect(container).toMatchSnapshot();
  });

  test("test square passing 0, should have class nothing", () => {
    const { getByTestId, container } = render(<Square count={0} />);
    const square = getByTestId("square-color");

    expect(square).toHaveClass("nothing");

    expect(container).toMatchSnapshot();
  });
});
