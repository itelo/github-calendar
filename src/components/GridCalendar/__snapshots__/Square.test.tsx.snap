// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Square /> test square passing 0, should have class nothing 1`] = `
<div>
  <div
    class="square-container"
  >
    <div
      class="square nothing"
      data-testid="square-color"
    />
  </div>
</div>
`;

exports[`<Square /> test square passing CONTRIBUTIONS_ALOT, should have class alot 1`] = `
<div>
  <div
    class="square-container"
  >
    <div
      class="square alot"
      data-testid="square-color"
    />
  </div>
</div>
`;

exports[`<Square /> test square passing CONTRIBUTIONS_LITTLE, should have class little 1`] = `
<div>
  <div
    class="square-container"
  >
    <div
      class="square little"
      data-testid="square-color"
    />
  </div>
</div>
`;

exports[`<Square /> test square passing CONTRIBUTIONS_MODERATE, should have class moderate 1`] = `
<div>
  <div
    class="square-container"
  >
    <div
      class="square moderate"
      data-testid="square-color"
    />
  </div>
</div>
`;

exports[`<Square /> test square passing CONTRIBUTIONS_MOST, should have class most 1`] = `
<div>
  <div
    class="square-container"
  >
    <div
      class="square most"
      data-testid="square-color"
    />
  </div>
</div>
`;
